### DISCUSSION TITLE
SSL Setup help

### DISCUSSION THREAD

**Discussion by TPSSCW** [chuot/rdio-scanner#316] [General]
Hi, I got the program installed and all my groups/channels created but i can only connect when using my home network. can someone help me with the ssl setup for connecting when away from home?

**Comment by PaTall** [comment]
Try Tailscale.  Works great.  Tailscale.com

**Reply by TPSSCW** [reply]
That appears that it would work for me but would also require all my listeners to download tailscale as well so probably not easy for them

**Comment by chuot** [comment]
This will require you to have a DNS entry somewhere that will point to your internet IP address. DYNDNS is one such service you can use.

Then  you configure your router to redirect incoming ports 80 and 443 to your local instance on port 3000 and 3001. So internet port 80 to internal port 3000 and internet port 443 to internal port 3001.

And you start Rdio Scanner with these arguments:

```bash
rdio-scanner -listen :3000 -ssl_listen :3001 -ssl_auto_cert your.dyndns.org
```

You can also add the `-config_save` argument to save these config values into the `rdio-scanner.ini` file so the next time you don't have to retype them everytime.

The `-ssl_auto_cert` argument is there to simplify the SSL certificate generation by using the Let's Encrypt automatic certificate. The only thing for it to work is that your Rdio Scanner must listen on port 80 and 443 on the internet side.

The PDF document that comes with the Rdio Scanner executable has some hints in that regards. You can go through the document to find some hints.
